core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/ComponentArchitectureBlankRefreshExtension.java

```diff
@@ -84,6 +89,20 @@ public void beforeRefresh(DDiagram diagram) {
 
     updateTargetDiagram(diagram, !contextualElements.isEmpty());
 
+    // -------------------------------------
+    // Remove all functions added directly on diagram.
+    // Fix for bug 560092, can be deleted in Capella 6.0
+    // -------------------------------------
+
+    String mappingName = MappingConstantsHelper.getMappingFunction(diagram);
+    NodeMapping mapping = DiagramServices.getDiagramServices().getNodeMapping(diagram, mappingName);
+    Collection<DDiagramElement> nodes = diagram.getOwnedDiagramElements().stream() //
+        .filter(AbstractDNode.class::isInstance) //
+        .filter(d -> d.getTarget() != null && d.getTarget() instanceof AbstractFunction) //

```

Comment:
`d.getTarget() != null` is not required since covered by ` d.getTarget() instanceof AbstractFunction`